pkg/driver/node/mounter/pod_mounter_test.go

```diff
@@ -59,9 +64,14 @@ func setup(t *testing.T) *testCtx {
 	kubeletPath := t.TempDir()
 	t.Setenv("KUBELET_PATH", kubeletPath)
 
-	// Chdir to `kubeletPath` so `mountoptions.{Recv, Send}` can use relative paths to Unix sockets
-	// to overcome `bind: invalid argument`.
-	t.Chdir(kubeletPath)
+	testCwdMu.Lock()
+	oldwd, err := os.Getwd()
+	assert.NoError(t, err)
+	assert.NoError(t, os.Chdir(kubeletPath))
+	t.Cleanup(func() {
+		_ = os.Chdir(oldwd)
+		testCwdMu.Unlock()
```

Comment:
Wouldn't unlocking the mutex in the `Cleanup` function, effectively serialize all tests execution? Since I believe the `Cleanup` function (https://pkg.go.dev/testing@master#T.Cleanup) is called when the test completes, that would hold the mutex for the entire duration of the test case.

Reply:
I do not want to change the non test code for now, before adding more E2E tests.

Reply:
I am thinking that using the `Ordered` flag may have had the same effect than the mutex, but I'm not sure

Reply:
yes it will, for unit tests we do not use ginkgo yet. And for 1 small thing I did not want to add a ginkgo depdenency